### DISCUSSION TITLE
Understanding KeyError with latest pull

### DISCUSSION THREAD

**Discussion by ganmad** [pybamm-team/PyBaMM#3369] [Q&A] [answered]
Hello Team,

After a recent pull of the latest changes from PyBaMM, my model (thermal+SEI+LAM) is throwing the following errors, which is used to work fine earlier. Can you help me understand where the problem is? 

----------------------------
pybamm/parameters/parameter_values.py:538

537 try:
--> 538     return self._processed_symbols[symbol]
    539 except KeyError:

===============================

KeyError: Division(0x411e4ca27e30ce6e, /, children=['z-average((Negative current collector thickness [m] * ((Current function [A] / (Number of electrodes connected in parallel to make a cell * Electrode width [m] * Electrode height [m])) ** 2.0) / Negative current collector conductivity [S.m-1] + Negative electrode thickness [m] * (x-average((broadcast(3.0 * X-averaged negative electrode active material volume fraction) / Negative particle radius [m]) * (broadcast(Current function [A] / (Number of electrodes connected in parallel to make a cell * Electrode width [m] * Electrode height [m]) / Negative electrode thickness [m] / x-average(broadcast(3.0 * X-averaged negative electrode active material volume fraction) / Negative particle radius [m])) - (exp((SEI growth activation energy [J.mol-1] / Ideal gas constant [J.K-1.mol-1]) * ((1.0 / Reference temperature [K]) - (1.0 / Volume-averaged cell temperature [K]))) * -Faraday constant [C.mol-1] * EC initial concentration in electrolyte [mol.m-3] * SEI kinetic rate constant [m.s-1] * exp(-0.5 * Faraday constant [C.mol-1] / (Ideal gas constant [J.K-1.mol-1] * Volume-averaged cell temperature [K]) * (broadcast(x-average((2.0 * Ideal gas constant [J.K-1.mol-1] * Volume-averaged cell temperature [K] / Faraday constant [C.mol-1]) * arcsinh(broadcast(Current function [A] / (Number of electrodes connected in parallel to make a cell * Electrode width [m] * Electrode height [m]) / Negative electrode thickness [m] / x-average(broadcast(3.0 * X-averaged negative electrode active material volume fraction) / Negative particle radius [m])) / (2.0 * Negative electrode exchange-current density [A.m-2]))) + Negative electrode OCP [V] + 1e-06 * (1.0 / (maximum(minimum(boundary value(X-averaged negative particle concentration [mol.m-3]) / Maximum concentration in negative electrode [mol.m-3], 0.9999999999), 1e-10)) + 1.0 / (-1.0 + maximum(minimum(boundary value(X-averaged negative particle concentration [mol.m-3]) / Maximum concentration in negative electrode [mol.m-3], 0.9999999999), 1e-10))) + (Volume-averaged cell temperature [K] - Reference temperature [K]) * Negative electrode OCP entropic change [V.K-1] - x-average(broadcast(-Current function [A] / (Number of electrodes connected in parallel to make a cell * Electrode width [m] * Electrode height [m]) / Negative electrode thickness [m] / x-average(broadcast(3.0 * X-averaged negative electrode active material volume fraction) / Negative particle radius [m])) * broadcast(X-averaged outer SEI thickness [m]) * SEI resistivity [Ohm.m]) - SEI open-circuit potential [V]) - (broadcast(Current function [A] / (Number of electrodes connected in parallel to make a cell * Electrode width [m] * Electrode height [m]) / Negative electrode thickness [m] / x-average(broadcast(3.0 * X-averaged negative electrode active material volume fraction) / Negative particle radius [m])) * broadcast(X-averaged outer SEI thickness [m]) * SEI resistivity [Ohm.m]))) / (1.0 + broadcast(X-averaged outer SEI thickness [m] / EC diffusivity [m2.s-1]) * SEI kinetic rate constant [m.s-1] * exp(-0.5 * Faraday constant [C.mol-1] / (Ideal gas constant [J.K-1.mol-1] * Volume-averaged cell temperature [K]) * (broadcast(x-average((2.0 * Ideal gas constant [J.K-1.mol-1] * Volume-averaged cell temperature [K] / Faraday constant [C.mol-1]) * arcsinh(broadcast(Current function [A] / (Number of electrodes connected in parallel to make a cell * Electrode width [m] * Electrode height [m]) / Negative electrode thickness [m] / x-average(broadcast(3.0 * X-averaged negative electrode active material volume fraction) / Negative particle radius [m])) / (2.0 * Negative electrode exchange-current density [A.m-2]))) + Negative electrode OCP [V] + 1e-06 * (1.0 / (maximum(minimum(boundary value(X-averaged negative particle concentration [mol.m-3]) / Maximum concentration in negative electrode [mol.m-3], 0.9999999999), 1e-10)) + 1.0 / (-1.0 + maximum(minimum(boundary value(X-averaged negative particle concentration [mol.m-3]) / Maximum concentration in negative electrode [mol.m-3], 0.9999999999), 1e-10))) + (Volume-averaged cell temperature [K] - Reference temperature [K]) * Negative electrode OCP entropic change [V.K-1] - x-average(broadcast(-Current function [A] / (Number of electrodes connected in parallel to make a cell * Electrode width [m] * Electrode height [m]) / Negative electrode thickness [m] / x-average(broadcast(3.0 * X-averaged negative electrode active material volume fraction) / Negative particle radius [m])) * broadcast(X-averaged outer SEI thickness [m]) * SEI resistivity [Ohm.m]) - SEI open-circuit potential [V]) - (broadcast(Current function [A] / (Number of electrodes connected in parallel to make a cell * Electrode width [m] * Electrode height [m]) / Negative electrode thickness [m] / x-average(broadcast(3.0 * X-averaged negative electrode active material volume fraction) / Negative particle radius [m])) * broadcast(X-averaged outer SEI thickness [m]) * SEI resistivity [Ohm.m])))))) * (2.0 * Ideal gas constant [J.K-1.mol-1] * Volume-averaged cell temperature [K] / Faraday constant [C.mol-1]) * arcsinh(broadcast(Current function [A] / (Number of electrodes connected in parallel to make a cell * Electrode width [m] * Electrode height [m]) / Negative electrode thickness [m] / x-average(broadcast(3.0 * X-averaged negative electrode active material volume fraction) / Negative particle radius [m])) / (2.0 * Negative electrode exchange-current density [A.m-2]))) + x-average((broadcast(3.0 * X-averaged negative electrode active material volume fraction) / Negative particle radius [m]) * (broadcast(Current function [A] / (Number of electrodes connected in parallel to make a cell * Electrode width [m] * Electrode height [m]) / Negative electrode thickness [m] / x-average(broadcast(3.0 * X-averaged negative electrode active material volume fraction) / Negative particle radius [m])) - (exp((SEI growth activation energy [J.mol-1] / Ideal gas constant [J.K-1.mol-1]) * ((1.0 / Reference temperature [K]) - (1.0 / Volume-averaged cell temperature [K]))) * -Faraday constant [C.mol-1] * EC initial concentration in electrolyte [mol.m-3] * SEI kinetic rate constant [m.s-1] * exp(-0.5 * Faraday constant [C.mol-1] / (Ideal gas constant [J.K-1.mol-1] * Volume-averaged cell temperature [K]) * (broadcast(x-average((2.0 * Ideal gas constant [J.K-1.mol-1] * Volume-averaged cell temperature [K] / Faraday constant [C.mol-1]) * arcsinh(broadcast(Current function [A] / (Number of electrodes connected in parallel to make a cell * Electrode width [m] * Electrode height [m]) / Negative electrode thickness [m] / x-average(broadcast(3.0 * X-averaged negative electrode active material volume fraction) / Negative particle radius [m])) / (2.0 * Negative electrode exchange-current density [A.m-2]))) + Negative electrode OCP [V] + 1e-06 * (1.0 / (maximum(minimum(boundary value(X-averaged negative particle concentration [mol.m-3]) / Maximum concentration in negative electrode [mol.m-3], 0.9999999999), 1e-10)) + 1.0 / (-1.0 + maximum(minimum(boundary value(X-averaged negative particle concentration [mol.m-3]) / Maximum concentration in negative electrode [mol.m-3], 0.9999999999), 1e-10))) + (Volume-averaged cell temperature [K] - Reference temperature [K]) * Negative electrode OCP entropic change [V.K-1] - x-average(broadcast(-Current function [A] / (Number of electrodes connected in parallel to make a cell * Electrode width [m] * Electrode height [m]) / Negative electrode thickness [m] / x-average(broadcast(3.0 * X-averaged negative electrode active material volume fraction) / Negative particle radius [m])) * broadcast(X-averaged outer SEI thickness [m]) * SEI resistivity [Ohm.m]) - SEI open-circuit potential [V]) - (broadcast(Current function [A] / (Number of electrodes connected in parallel to make a cell * Electrode width [m] * Electrode height [m]) / Negative electrode thickness [m] / x-average(broadcast(3.0 * X-averaged negative electrode active material volume fraction) / Negative particle radius [m])) * broadcast(X-averaged outer SEI thickness [m]) * SEI resistivity [Ohm.m]))) / (1.0 + broadcast(X-averaged outer SEI thickness [m] / EC diffusivity [m2.s-1]) * SEI kinetic rate constant [m.s-1] * exp(-0.5 * Faraday constant [C.mol-1] / (Ideal gas constant [J.K-1.mol-1] * Volume-averaged cell temperature [K]) * (broadcast(x-average((2.0 * Ideal gas constant [J.K-1.mol-1] * Volume-averaged cell temperature [K] / Faraday constant [C.mol-1]) * arcsinh(broadcast(Current function [A] / (Number of electrodes connected in parallel to make a cell * Electrode width [m] * Electrode height [m]) / Negative electrode thickness [m] / x-average(broadcast(3.0 * X-averaged negative electrode active material volume fraction) / Negative particle radius [m])) / (2.0 * Negative electrode exchange-current density [A.m-2]))) + Negative electrode OCP [V] + 1e-06 * (1.0 / (maximum(minimum(boundary value(X-averaged negative particle concentration [mol.m-3]) / Maximum concentration in negative electrode [mol.m-3], 0.9999999999), 1e-10)) + 1.0 / (-1.0 + maximum(minimum(boundary value(X-averaged negative particle concentration [mol.m-3]) / Maximum concentration in negative electrode [mol.m-3], 0.9999999999), 1e-10))) + (Volume-averaged cell temperature [K] - Reference temperature [K]) * Negative electrode OCP entropic change [V.K-1] - x-average(broadcast(-Current function [A] / (Number of electrodes connected in parallel to make a cell * Electrode width [m] * Electrode height [m]) / Negative electrode thickness [m] / x-average(broadcast(3.0 * X-averaged negative electrode active material volume fraction) / Negative particle radius [m])) * broadcast(X-averaged outer SEI thickness [m]) * SEI resistivity [Ohm.m]) - SEI open-circuit potential [V]) - (broadcast(Current function [A] / (Number of electrodes connected in parallel to make a cell * Electrode width [m] * Electrode height [m]) / Negative electrode thickness [m] / x-average(broadcast(3.0 * X-averaged negative electrode active material volume fraction) / Negative particle radius [m])) * broadcast(X-averaged outer SEI thickness [m]) * SEI resistivity [Ohm.m])))))) * Volume-averaged cell temperature [K] * broadcast(Negative electrode OCP entropic change [V.K-1]))) + Positive electrode thickness [m] * (x-average((broadcast(3.0 * X-averaged positive electrode active material volume fraction) / Positive particle radius [m]) * broadcast(-Current function [A] / (Number of electrodes connected in parallel to make a cell * Electrode width [m] * Electrode height [m]) / Positive electrode thickness [m] / x-average(broadcast(3.0 * X-averaged positive electrode active material volume fraction) / Positive particle radius [m])) * broadcast((2.0 * Ideal gas constant [J.K-1.mol-1] * Volume-averaged cell temperature [K] / Faraday constant [C.mol-1]) * arcsinh(-Current function [A] / (Number of electrodes connected in parallel to make a cell * Electrode width [m] * Electrode height [m]) / Positive electrode thickness [m] / x-average(broadcast(3.0 * X-averaged positive electrode active material volume fraction) / Positive particle radius [m]) / (2.0 * Positive electrode exchange-current density [A.m-2])))) + x-average((broadcast(3.0 * X-averaged positive electrode active material volume fraction) / Positive particle radius [m]) * broadcast(-Current function [A] / (Number of electrodes connected in parallel to make a cell * Electrode width [m] * Electrode height [m]) / Positive electrode thickness [m] / x-average(broadcast(3.0 * X-averaged positive electrode active material volume fraction) / Positive particle radius [m])) * Volume-averaged cell temperature [K] * broadcast(Positive electrode OCP entropic change [V.K-1]))) + Positive current collector thickness [m] * ((Current function [A] / (Number of electrodes connected in parallel to make a cell * Electrode width [m] * Electrode height [m])) ** 2.0) / Positive current collector conductivity [S.m-1]) / (Negative current collector thickness [m] + Negative electrode thickness [m] + Separator thickness [m] + Positive electrode thickness [m] + Positive current collector thickness [m])) + (-Total heat transfer coefficient [W.m-2.K-1] * Cell cooling surface area [m2] / Cell volume [m3]) * (Volume-averaged cell temperature [K] - z-average(Ambient temperature [K]))', '(Negative current collector density [kg.m-3] * Negative current collector specific heat capacity [J.kg-1.K-1] * Negative current collector thickness [m] + Negative electrode density [kg.m-3] * Negative electrode specific heat capacity [J.kg-1.K-1] * Negative electrode thickness [m] + Separator density [kg.m-3] * Separator specific heat capacity [J.kg-1.K-1] * Separator thickness [m] + Positive electrode density [kg.m-3] * Positive electrode specific heat capacity [J.kg-1.K-1] * Positive electrode thickness [m] + Positive current collector density [kg.m-3] * Positive current collector specific heat capacity [J.kg-1.K-1] * Positive current collector thickness [m]) / (Negative current collector thickness [m] + Negative electrode thickness [m] + Separator thickness [m] + Positive electrode thickness [m] + Positive current collector thickness [m])'], domains={})

ValueError: operands could not be broadcast together with shapes (3,1) (33,1) 

==============================================

Its not showing that any parameter is missing. Is it because of some zero parameter or non-existent parameter?

Thanks,

**Comment by brosaplanella** [comment]
Can you share a minimum working example that we can run on our end?

**Reply by ganmad** [reply, marked as answer]
Hi @brosaplanella,

Thanks for the response. I figured out that temperature dependent SEI diffusivity I defined locally on a notebook in the old file is causing the error with the latest updates. I removed it and it is working fine. Thanks. 

Regards,
Ganesh
